src/view.cpp

```diff
@@ -444,18 +445,26 @@ void View::push_searchresult(std::shared_ptr<RssFeed> feed,
 {
 	assert(feed != nullptr);
 	LOG(Level::DEBUG, "View::push_searchresult: pushing search result");
-
 	if (feed->total_item_count() > 0) {
-		auto searchresult = std::make_shared<ItemListFormAction>(
-				this, itemlist_str, rsscache, filters, cfg, rxman);
-		searchresult->set_feed(feed);
-		searchresult->set_show_searchresult(true);
-		searchresult->set_searchphrase(phrase);
-		apply_colors(searchresult);
-		searchresult->set_parent_formaction(get_current_formaction());
-		searchresult->init();
-		formaction_stack.push_back(searchresult);
-		current_formaction = formaction_stack_size() - 1;
+		if (this->get_current_formaction()->id() != "searchresultslist") {
+			auto searchresult = std::make_shared<SearchResultsFormAction>(
+					this, itemlist_str, rsscache, filters, cfg, rxman);
+			searchresult->set_feed(feed);
+			searchresult->set_show_searchresult(true);
+			searchresult->set_searchphrase(phrase);
+			apply_colors(searchresult);
+			searchresult->set_parent_formaction(get_current_formaction());
+			searchresult->add_history(feed);
+			searchresult->init();
+			formaction_stack.push_back(searchresult);
+			current_formaction = formaction_stack_size() - 1;
+		} else {
+			auto searchresult = std::static_pointer_cast<SearchResultsFormAction>
+				(this->get_current_formaction());
+			searchresult->set_feed(feed);
+			searchresult->add_history(feed);
+			searchresult->set_searchphrase(phrase);
```

Comment:
I think these three methods should be combined into a single `add_to_history()` method, because they are always called simultaneously.

Reply:
@Minoru I disagree, I think `set_feed` and `set_searchphrase`, can't be a part of area of responsibility of `add_to_history`

Reply:
Why?

Note I called my method `add_to_history()`. The existing `add_history()` is a setter, so yeah, I agree it has a very small area of responsibility. But I think it's warranted to create an entirely new method, `add_to_history()`, which will combine the three and avoid temporal coupling between them. (`add_history()` can then be removed, because it won't be used anywhere anymore.)

Reply:
Ping.

Reply:
@Minoru Done.